routers/web/projects/workflows.go

```diff
@@ -0,0 +1,624 @@
+}
+
+type WorkflowsPostForm struct {
+	EventID string         `json:"event_id"`
+	Filters map[string]any `json:"filters"`
+	Actions map[string]any `json:"actions"`
+}
+
+// WorkflowsPost handles creating or updating a workflow
+func WorkflowsPost(ctx *context.Context) {
+	projectID := ctx.PathParamInt64("id")
+	p, err := project_model.GetProjectByID(ctx, projectID)
+	if err != nil {
+		if project_model.IsErrProjectNotExist(err) {
+			ctx.NotFound(nil)
+		} else {
+			ctx.ServerError("GetProjectByID", err)
+		}
+		return
+	}
+	if p.Type == project_model.TypeRepository && p.RepoID != ctx.Repo.Repository.ID {
+		ctx.NotFound(nil)
+		return
+	}
+	if (p.Type == project_model.TypeOrganization || p.Type == project_model.TypeIndividual) && p.OwnerID != ctx.ContextUser.ID {
+		ctx.NotFound(nil)
+		return
+	}
+
+	// Handle both form data and JSON data
+	// Handle JSON data
+	form := &WorkflowsPostForm{}
+	content, err := io.ReadAll(ctx.Req.Body)
+	if err != nil {
+		ctx.ServerError("ReadRequestBody", err)
+		return
+	}
+	defer ctx.Req.Body.Close()
+	log.Trace("get " + string(content))
+	if err := json.Unmarshal(content, &form); err != nil {
+		ctx.ServerError("DecodeWorkflowsPostForm", err)
+		return
+	}
+	if form.EventID == "" {
+		ctx.JSON(http.StatusBadRequest, map[string]any{"error": "InvalidEventID", "message": "EventID is required"})
+		return
+	}
+
+	// Convert form data to filters and actions
+	filters := convertFormToFilters(form.Filters)
+	actions := convertFormToActions(form.Actions)
+
+	// Validate: at least one action must be configured
+	if len(actions) == 0 {
+		ctx.JSON(http.StatusBadRequest, map[string]any{
+			"error":   "NoActions",
+			"message": ctx.Tr("projects.workflows.error.at_least_one_action"),
+		})
+		return
+	}
+
+	eventID, _ := strconv.ParseInt(form.EventID, 10, 64)
+	if eventID == 0 {
+		// check if workflow event is valid
+		if !project_model.IsValidWorkflowEvent(form.EventID) {
+			ctx.JSON(http.StatusBadRequest, map[string]any{"error": "EventID is invalid"})
+			return
+		}
+
+		// Create a new workflow for the given event
+		wf := &project_model.Workflow{
+			ProjectID:       projectID,
+			WorkflowEvent:   project_model.WorkflowEvent(form.EventID),
+			WorkflowFilters: filters,
+			WorkflowActions: actions,
+			Enabled:         true, // New workflows are enabled by default
+		}
+		if err := project_model.CreateWorkflow(ctx, wf); err != nil {
+			ctx.ServerError("CreateWorkflow", err)
+			return
+		}
+
+		// Return the newly created workflow with filter summary
+		workflowSummary := project_service.GetWorkflowSummary(ctx, wf)
+		ctx.JSON(http.StatusOK, map[string]any{
+			"success": true,
+			"workflow": map[string]any{
+				"id":           wf.ID,
+				"event_id":     strconv.FormatInt(wf.ID, 10),
+				"display_name": string(ctx.Tr(wf.WorkflowEvent.LangKey())),
+				"filters":      wf.WorkflowFilters,
+				"actions":      wf.WorkflowActions,
+				"summary":      workflowSummary,
+				"enabled":      wf.Enabled,
+			},
```

Comment:
No struct for this? It looks the same as `WorkflowConfig`?

Reply:
48e144d015